### ISSUE TITLE
lower accuracy in 2.2.0 than in 1.2.0

### ISSUE THREAD

**Issue by Ch3nYe** [OpenNMT/OpenNMT-py#2148]
I use the same configuration and the same courpus data train on 2.2.0 and 1.2.0, but the former's performance(25%) is lower than the latter (30%).
there is the configuration:
on 2.2.0 :
**step1**
`onmt_build_vocab -config data_preprocess-len1500-2.yaml`
```yaml
# dataset save
save_data: summarizational_processed/data-len1500-2
# vocab save
src_vocab: summarizational_processed/data-len1500-2.vocab.src
tgt_vocab: summarizational_processed/data-len1500-2.vocab.tgt
src_vocab_size: 400000
tgt_vocab_size: 40000
src_seq_length: 3000
tgt_seq_length: 30

overwrite: True

# corpus source path
data:
    corpus:
        path_src: train/src-train-len1500-2.txt
        path_tgt: train/tgt-train-len1500-2.txt
    valid:
        path_src: valid/src-valid-len1500-2.txt
        path_tgt: valid/tgt-valid-len1500-2.txt

# opt
num_threads: 10
n_sample: -1
```
**step2**
`onmt_train -config train_len1500-2_wo_embedding_BiLSTM.yaml`
```yaml
data:
    corpus:
        path_src: train/src-train-len1500.txt
        path_tgt: train/tgt-train-len1500.txt
    valid:
        path_src: valid/src-valid-len1500.txt
        path_tgt: valid/tgt-valid-len1500.txt

src_vocab: summarizational_processed/data-len1500.vocab.src
tgt_vocab: summarizational_processed/data-len1500.vocab.tgt
src_vocab_size: 400000
tgt_vocab_size: 40000

# train
save_model: summarizational_model/model_wo_embedding_BiLSTM
save_checkpoint_steps: 50000
keep_checkpoint: 20
seed: 233
train_steps: 1000000
valid_steps: 50000
# warmup_steps: 8000
report_every: 1000
master_port: 2333
log_file: summarizational_model/train_wo_embedding_BiLSTM.log

# model
decoder_type: rnn
encoder_type: brnn
word_vec_size: 128
rnn_size: 128

accum_count: [1]
optim: adam
adam_beta1: 0.9 # default
adam_beta2: 0.999 
learning_rate_decay: 1
learning_rate: 0.001

batch_size: 64
valid_batch_size: 64
batch_type: sents # default
normalization: sents # default
dropout: 0.3 # default
label_smoothing: 0.0 # default

# max_generator_batches: 2

param_init: 0.0
# param_init_glorot: 'true'
# position_encoding: 'true'

world_size: 2
gpu_ranks:
- 0
- 1
```

on 1.2.0 :
**step1**
```shell
onmt_preprocess \
-train_src train/src-train-len1500-2.txt \
-train_tgt train/tgt-train-len1500-2.txt \
-valid_src valid/src-valid-len1500-2.txt \
-valid_tgt valid/tgt-valid-len1500-2.txt \
-save_data summarizational_processed/data-len1500-2 \
-src_vocab_size 200000 \
-tgt_vocab_size 40000 \
-src_seq_length 3000 \
-tgt_seq_length 50 \
-num_threads 1
```
**step2**
```shell
onmt_train \
-data summarizational_processed/data-len1500-2 \
-save_model summarizational_model/model_1.2.0_wo_len1500-2_BiLSTM \
-world_size 1 -gpu_ranks 0 -word_vec_size 128 \
-train_steps 1000000 -learning_rate_decay 1 \
-optim adam -learning_rate 0.001 \
-save_checkpoint_steps 50000 \
-batch_size 64 \
-valid_steps 50000 -valid_batch_size 32 \
-log_file summarizational_model/train_1.2.0_wo_len1500-2_BiLSTM.log
```

**Did I miss something different between these configurations above?** In my view, the same data: {tgt,src}-{train valid}-len1500-2.txt, the same model: brnn-rnn namely BiLSTM. Hope you guys help me check it...

**Comment by francoishernandez** [comment]
A few mismatches in your configs :
`src/tgt_seq_length` is used by the `filtertoolong` transform, which is not set in your config
`src_vocab_size` is different in your runs. A small diff might not have an impact but here 200k vs 400k might.
By the way these vocab sizes seem quite big.

**Comment by Ch3nYe** [comment]
> A few mismatches in your configs : `src/tgt_seq_length` is used by the `filtertoolong` transform, which is not set in your config `src_vocab_size` is different in your runs. A small diff might not have an impact but here 200k vs 400k might. By the way these vocab sizes seem quite big.

Thanks, 
1. "len1500" just a tag in file name, it means that I already truncate the src data to 1500 token each line. So I think that is equivalent to using `filtertoolong` transform.
2. Yes, `src_vocab_size ` is different, I will check it.

**Comment by Ch3nYe** [comment]
I already try set the same `src_vocab_size ` in 220, but the acc is not change significantly (about 0.5%).
May I open this issue? maybe there are other people meeting the same problem.
